Sources/ClientRuntime/Networking/Http/URLSession/FoundationStreamBridge.swift

```diff
@@ -44,80 +50,99 @@ class FoundationStreamBridge: NSObject, StreamDelegate {
     /// A Logger for logging events.
     private let logger: LogAgent
 
-    /// Actor used to ensure writes are performed in series.
-    actor WriteCoordinator {
+    /// Actor used to ensure writes are performed in series, one at a time.
+    private actor WriteCoordinator {
         var task: Task<Void, Error>?
 
-        /// `true` if the readable stream has been found to be empty, `false` otherwise.  Will flip to `true` if the readable stream is read,
-        /// and `nil` is returned.
-        var readableStreamIsEmpty = false
-
-        /// Sets stream status to indicate the stream is empty.
-        func setReadableStreamIsEmpty() async {
-            readableStreamIsEmpty = true
-        }
-
         /// Creates a new concurrent Task that executes the passed block, ensuring that the previous Task
         /// finishes before this task starts.
         ///
         /// Acts as a sort of "serial queue" of Swift concurrency tasks.
         /// - Parameter block: The code to be performed in this task.
-        func perform(_ block: @escaping @Sendable (WriteCoordinator) async throws -> Void) {
-            self.task = Task { [task] in
+        func perform(_ block: @escaping @Sendable () async throws -> Void) async throws {
+            let task = Task { [task] in
                 _ = await task?.result
-                try await block(self)
+                try await block()
             }
+            self.task = task
+            _ = try await task.value
```

Comment:
Simplified this method, and (on this line) made it rethrow any error that was thrown by its `Task`.